src/EventListener/StoreInDoctrineOnMessageSentListener.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace KaroIO\MessengerMonitorBundle\EventListener;
+
+use KaroIO\MessengerMonitorBundle\Storage\StoredMessage;
+use KaroIO\MessengerMonitorBundle\Storage\StoredMessageRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\SendMessageToTransportsEvent;
+
+/**
+ * @internal
+ */
+final class StoreInDoctrineOnMessageSentListener implements EventSubscriberInterface
+{
+    private $storedMessageRepository;
+
+    public function __construct(StoredMessageRepository $storedMessageRepository)
+    {
+        $this->storedMessageRepository = $storedMessageRepository;
+    }
+
+    public function onMessageSent(SendMessageToTransportsEvent $event): void
+    {
+        $this->storedMessageRepository->saveMessage(StoredMessage::fromEnvelope($event->getEnvelope()));
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            // should happen after AddStampOnMessageSentListener
+            SendMessageToTransportsEvent::class => ['onMessageSent', 1],
```

Comment:
Because there are various different listeners that need a specific ordering, it might be better to use 10-based priorities - like 10 and 20 instead of 1 and 2. It adds some "flexibility" that a user could add a listener *between* 10 and 20 - using 1 and 2 doesn't allow that.